### DISCUSSION TITLE
How to use Headers in `useFetch`?

### DISCUSSION THREAD

**Discussion by arun-mani-j** [nuxt/nuxt#17113] [Questions] [answered]
Hii.
I'm pretty new to web development itself, so please bear with my noobines :sweat_smile: .

I would like to know how to use headers in a `useFetch` call.

This is my `app.vue`:
```vue
<template>
  <div>
    {{ data }}
  </div>
</template>

<script setup>
const apiEndPoint = "http://127.0.0.1:8000";
const options = {
  key: new Date().toISOString(),
  headers: [{ key: "Authorization", value: `Bearer 123` }],
};

const { data } = await useFetch(apiEndPoint + "/test/", options);
</script>
```

This is my `server.py` run at `http://127.0.0.1:8000`:
```python
"""Main FastAPI instance."""
from fastapi import FastAPI, Request
from fastapi.middleware.cors import CORSMiddleware

origins = [
    "http://localhost:3000",
]

app = FastAPI(redoc_url=None)


app.add_middleware(
    CORSMiddleware,
    allow_origins=origins,
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)


@app.get("/test/")
async def test(req: Request):
    print(req.headers.get("Authorization", "NOT GIVEN"))
    return 1
```

If I run both the above apps, I don't get `1` displayed on my website (`/test/` returns `1`). However, if I comment out `headers: ...` line in `app.vue` everything works fine but in FastAPI app's console, I get `NOT GIVEN` printed.

If I include the headers, the API endpoint is not even called (FastAPI logs all the API calls).

Everything works good in `curl`:
```bash
$ curl -H 'Authorization: u' -X GET 'http://127.0.0.1:8000/test/'
1
```
I'm using Firefox 103.0.1 on Debian GNU/Linux.
What is the mistake I'm doing here? Thanks in advance!

**Comment by ian4uia** [comment, marked as answer]
This is a snippet from my code which is working.

```
let authData = {}
if (catStore.authUser !== null) {
  const jwt = sign({ user: catStore.authUser, role: catStore.authRole }, catStore.secret)
  console.log(jwt)
  authData = { Authorization: `Bearer ${jwt}` }
}
const { data: audits, pending, refresh } = useLazyAsyncData(
  'audits',
  () => $fetch('/api/audits', {
    headers: authData
  }),
  { server: true }
)
```
So I think your code should be:
`headers: { Authorization:  'Bearer 123' }`

**Reply by arun-mani-j** [reply]
Wow thanks! It is working!

In the docs, we have `headers?: { key: string, value: string }[]`, doesn't the `[]` mean a list though? :thinking:
